llama-index-core/llama_index/core/workflow/events.py

```diff
@@ -132,6 +134,20 @@ def custom_model_dump(self, handler: Any) -> Dict[str, Any]:
             data["_data"] = self._data
         return data
 
+    def _generate_run_id(self) -> str:
+        return str(uuid.uuid4())
+
+    def _assign_new_run_id(self) -> None:
+        self.run_id = self._generate_run_id()
+
+    def _assign_same_run_id_as(self, ev: "Event") -> None:
+        self.run_id = ev.run_id
+
+    def _copy_with_new_run_id(self) -> "Event":
+        copy = self.model_copy(deep=True)
+        copy.run_id = self._generate_run_id()
+        return copy
```

Comment:
Does it make more sense to attach the run id to the context?

Reply:
closing, as no longer relevant with recent changes